website/client/components/challenges/challengeModal.vue

```diff
@@ -233,7 +233,10 @@ export default {
   async mounted () {
     this.groups = await this.$store.dispatch('guilds:getMyGuilds');
```

Comment:
we could move this code from the mounted event that is called every time to only when the modal is actually shown for example by calling a method when the modal is shown. You can attach a method to the shown event by adding `@shown="methodName"` at line 2 in this file

Reply:
more info here https://bootstrap-vue.js.org/docs/components/modal

Reply:
I made and pushed this change to the PR, but it isn't finished until I resolve the first issue you commented on. But this should work as proof of concept - the API call isn't made until the modal is shown. Thanks for the input - I knew there had to be something like this (and I should have just looked it up!)